src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2123,15 +2123,32 @@ namespace cryptonote
     };
     typedef epee::misc_utils::struct_init<request_t> request;
 
+    struct entry
+    {
+      uint8_t hf_version;
+      uint64_t height;
+
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(hf_version)
+        KV_SERIALIZE(height)
+      END_KV_SERIALIZE_MAP()
+    };
+
     struct response_t: public rpc_response_base
     {
       uint32_t version;
       bool release;
+      uint64_t current_height;
+      uint64_t target_height;
+      std::vector<entry> hard_forks;
 
       BEGIN_KV_SERIALIZE_MAP()
         KV_SERIALIZE_PARENT(rpc_response_base)
         KV_SERIALIZE(version)
         KV_SERIALIZE(release)
+        KV_SERIALIZE(current_height)
+        KV_SERIALIZE(target_height)
+        KV_SERIALIZE(hard_forks)
```

Comment:
`COMMAND_RPC_GET_VERSION::response_t` packets serialized by pre-PR daemons will not be able to be deserialized by post-PR wallets (`node_rpc_proxy` specifically). I would supply some optional values through `KV_SERIALIZE_OPT` to maintain backwards compatibility.

Reply:
> I would supply some optional values through KV_SERIALIZE_OPT

I agree this seems safer, see the latest.

FWIW before this change, I had tested a wallet running the PR's code pointing to pre-PR daemons and vice versa; it worked fine deserializing with 0 values thanks to [this](https://github.com/monero-project/monero/blob/6402dbee69867c76530b45e8a21b692850fea03f/src/wallet/node_rpc_proxy.cpp#L98). But I prefer handling it more explicitly as you suggested.